### PR TITLE
[Merged by Bors] - Support benchmark about model optimization

### LINKED ISSUE #6: Support benchmark about model optimization

### PR DESCRIPTION
**Pull request by priscilla-pan**
Closes #6

### CODE REVIEW

File: benchmark/tests/client_script/resnet50_client_script.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tests/client_script/resnet50_client_script.sh:1]: Use "#!/bin/bash -e"

File: benchmark/tests/client_script/resnet50_client_script.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+python3 /home/john/Adlik/benchmark/tests/client/$CLIENT_INFERENCE_SCRIPT --model-name=resnet50  --batch-size=100 /home/john/Adlik/benchmark/tests/data/$IMAGE_FILENAME && \
```
Comment (shiyingjin) [inline benchmark/tests/client_script/resnet50_client_script.sh:2]: Don't use path like "/home/jonh/.....". We can't assume the user's work path.

File: benchmark/tests/client_script/resnet50_pruned_client_script.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tests/client_script/resnet50_pruned_client_script.sh:1]: Use "#!/bin/bash -e"

File: benchmark/tests/serving_script/resnet50_pruned_serving_script.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tests/serving_script/resnet50_pruned_serving_script.sh:1]: Use "#!/bin/bash -e"

File: benchmark/tests/serving_script/resnet50_serving_script.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tests/serving_script/resnet50_serving_script.sh:1]: Use "#!/bin/bash -e"

File: benchmark/tools/resnet50/automatic_test.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test.sh:1]: Use "#!/usr/bin/env bash -e"

File: benchmark/tools/resnet50/automatic_test_pruned.sh [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test_pruned.sh:1]: Use "#!/usr/bin/env bash -e"

File: benchmark/tools/resnet50/automatic_test_pruned.sh [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+cd ../../../../
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test_pruned.sh]: Where am I?

File: benchmark/tools/resnet50/automatic_test.sh [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+cd ../../../../
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test.sh]: Where am I?

File: benchmark/tests/client_script/resnet50_client_script.sh (hunk 2 above)
Comment (shiyingjin) [inline benchmark/tests/client_script/resnet50_client_script.sh:2]: Use double quotes to enclose parameters

File: benchmark/tests/client_script/resnet50_pruned_client_script.sh [hunk 10]
```diff
@@ -0,0 +1,4 @@
...
+python3 /home/john/Adlik/benchmark/tests/client/$CLIENT_INFERENCE_SCRIPT --model-name=resnet50_pruned  --batch-size=100 /home/john/Adlik/benchmark/tests/data/$IMAGE_FILENAME && \
```
Comment (shiyingjin) [inline benchmark/tests/client_script/resnet50_pruned_client_script.sh:2]: Use double quotes to enclose parameters

File: benchmark/tools/resnet50/automatic_test.sh [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+git clone https://github.com/Adlik/Adlik.git
+base_path=$(cd `dirname $0`; pwd)
+echo "base_path: "$base_path
+optimizer_benchmark_path=$base_path/model_optimizer/benchmark
+adlik_benchmark_path=$base_path/Adlik/benchmark
+adlik_path=$base_path/Adlik 
+cp -r $optimizer_benchmark_path/tests $adlik_benchmark_path
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test.sh:11]: Use double quotes to enclose parameters

File: benchmark/tools/resnet50/automatic_test.sh [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+cd $adlik_path
+echo "*****************start to make benchmark image and test****************************"
+python3 benchmark/src/automatic_test.py -d benchmark/tests/docker_test/tensorflow_gpu.Dockerfile -s tensorflow_gpu -b . -a . -m resnet50  -c benchmark/tests/client_script/resnet50_client_script.sh -ss benchmark/tests/serving_script/resnet50_serving_script.sh -l /home/pjy/adlik/Adlik/benchmark/log -tm benchmark/tests/test_model/resnet50_keras -cis imagenet_client.py -i imagenet.JPEG -cs benchmark/tests/compile_script/compile_script.sh  -sj resnet50_serving_model.json -gl 0
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test.sh:15]: The parameters of "automatic_test.py" are hard to understand

File: benchmark/tools/resnet50/automatic_test_pruned.sh [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+git clone https://github.com/Adlik/Adlik.git
+base_path=$(cd `dirname $0`; pwd)
+echo "base_path: "$base_path
+optimizer_benchmark_path=$base_path/model_optimizer/benchmark
+adlik_benchmark_path=$base_path/Adlik/benchmark
+adlik_path=$base_path/Adlik 
+cp -r $optimizer_benchmark_path/tests $adlik_benchmark_path
```
Comment (shiyingjin) [inline benchmark/tools/resnet50/automatic_test_pruned.sh:14]: Use double quotes to enclose parameters

**Review by shiyingjin — CHANGES_REQUESTED** [review]

**Review by shiyingjin — APPROVED** [review]

**Review by ChengcanWang-zte — APPROVED** [review]

File: benchmark/tools/resnet50/automatic_test.sh (hunk 9 above)
Comment (amadeus-zte) [reply benchmark/tools/resnet50/automatic_test.sh]: There is a hypothesis that current path of this runtime env is ‘xxxx’. 
It is not acceptable for users. Please give some general solutions.

**Pull request closed without merging**